Source/WebCore/testing/Internals.cpp

```diff
@@ -2238,6 +2236,9 @@ ExceptionOr<String> Internals::viewBaseBackgroundColor()
     Document* document = contextDocument();
     if (!document || !document->view())
         return Exception { ExceptionCode::InvalidAccessError };
+
+    document->updateLayout(LayoutOptions::IgnorePendingStylesheets);
```

Comment:
Because there are tests that change dark mode then fetch `baseBackgroundColor` immediately; we need this to compute the new color. Other `Internals` functions do this.

Reply:
I'm confused why we need to update layout though, shouldn't updating style be sufficient?

Reply:
We've been bitten before by functions we don't expect to perform layout to perform layout (I helped Brady with one in the past)

Reply:
All our API entry points that need up-to-date style use `updateLayout()`; we rarely (never?) just update style.